test/functional/tests/failover_standby/test_failover_multihost.py

```diff
@@ -0,0 +1,534 @@
+
+    postfailover_check(primary_node, core_path, original_core_md5, original_cache_stats)
+
+    with TestRun.use_dut(secondary_node):
+        TestRun.executor.wait_for_connection()
+
+    TestRun.LOGGER.end_group()
+
+
+def check_drbd_installed(duts):
+    for dut in duts:
+        with TestRun.use_dut(dut):
+            if not Drbd.is_installed():
+                TestRun.fail(f"DRBD is not installed on DUT {dut.ip}")
+
+
+def prepare_devices(duts):
+    for dut in duts:
+        with TestRun.use_dut(dut):
+            TestRun.dut.hostname = TestRun.executor.run_expect_success("uname -n").stdout
+
+            raid_members = [TestRun.disks["raid_dev1"], TestRun.disks["raid_dev2"]]
+            for d in raid_members:
+                d.create_partitions([raid_size * 1.1])  # extra space for RAID metadata
+
+            raid_config = RaidConfiguration(
+                level=Level.Raid1,
+                metadata=MetadataVariant.Legacy,
+                number_of_devices=2,
+                size=raid_size,
+            )
+            dut.raid = Raid.create(raid_config, [d.partitions[0] for d in raid_members])
+            dut.raid_path = readlink(dut.raid.path)
+
+            TestRun.disks["metadata_dev"].create_partitions([metadata_size] * 2)
+            dut.cache_md_dev = TestRun.disks["metadata_dev"].partitions[0]
+            dut.core_md_dev = TestRun.disks["metadata_dev"].partitions[1]
+
+            TestRun.disks["core_dev"].create_partitions([core_size])
+            dut.core_dev = TestRun.disks["core_dev"].partitions[0]
+
+
+def get_drbd_configs(n1, n2):
+    cache_original_drbd_nodes = [
+        Node(n1.hostname, n1.raid_path, n1.cache_md_dev.path, n1.ip, "7790"),
+        Node(n2.hostname, cache_exp_obj_path, n2.cache_md_dev.path, n2.ip, "7790"),
+    ]
+    cache_failover_drbd_nodes = [
+        Node(n1.hostname, cache_exp_obj_path, n1.cache_md_dev.path, n1.ip, "7790"),
+        Node(n2.hostname, n2.raid_path, n2.cache_md_dev.path, n2.ip, "7790"),
+    ]
+    core_drbd_nodes = [
+        Node(dut.hostname, dut.core_dev.path, dut.core_md_dev.path, dut.ip, "7791")
+        for dut in [n1, n2]
+    ]
+
+    caches_original_resource = Resource(
+        name="caches", device="/dev/drbd0", nodes=cache_original_drbd_nodes
+    )
+    caches_failover_resource = Resource(
+        name="caches", device="/dev/drbd0", nodes=cache_failover_drbd_nodes
+    )
+    cores_resource = Resource(name="cores", device="/dev/drbd100", nodes=core_drbd_nodes)
+
+    return caches_original_resource, caches_failover_resource, cores_resource
+
+
+def power_failure(primary_node, core_path):
+    with TestRun.use_dut(primary_node), TestRun.step("Make sure drbd instances are in sync"):
+        primary_node.cache_drbd.wait_for_sync()
+        primary_node.core_drbd.wait_for_sync()
+
+    with TestRun.use_dut(primary_node), TestRun.step("Get cache stats and core's md5"):
+        core_md5 = TestRun.executor.run(f"md5sum {core_path}").stdout.split()[0]
```

Comment:
I think in order to make the test execution more predictable I would suggest to switch cache to WO mode for the time of checksum calculation (and switch back after calculation). Currently after failover md5 calculation results in multiple partial misses and the cache is cleaned as a result. This isn't bad, but it is unexpected.

This applies to this line as well as the other where md5 is calculated.